guides/v2.2/magento-functional-testing-framework/release-2/extending.md

```diff
@@ -0,0 +1,356 @@
+        </assertCount>
+    </actionGroup>
+
+    <actionGroup name="CountProductB" extends="CountProductA">
+        <grabMultiple selector="selectorForProductB" stepKey="grabProducts"/>
+    </actionGroup>
+</actionGroups>
+```
+
+The new action groups correspond to:
+
+```xml
+<actionGroups>
+    <actionGroup name="CountProductA">
+        <arguments>
+            <argument name="count" type="string"/>
+        </arguments>
+        <grabMultiple selector="selectorForProductA" stepKey="grabProducts"/>
+        <assertCount stepKey="assertCount">
+            <expectedResult type="int">{{count}}</expectedResult>
+            <actualResult type="variable">grabProducts</actualResult>
+        </assertCount>
+    </actionGroup>
+
+    <actionGroup name="CountProductB">
+        <arguments>
+            <argument name="count" type="string"/>
+        </arguments>
+        <grabMultiple selector="selectorForProductB" stepKey="grabProducts"/>
+        <assertCount stepKey="assertCount">
+            <expectedResult type="int">{{count}}</expectedResult>
+            <actualResult type="variable">grabProducts</actualResult>
+        </assertCount>
+    </actionGroup>
+</actionGroups>
+```
+
+### Add an action
+
+**Use case**: The `grabProducts` action group needs count the products grabbed after
+Modify the action group to add steps for counting the object:
+
+```xml
+<actionGroups>
+    <actionGroup name="GetProductCount">
+        <arguments>
+            <argument name="productSelector" type="string"/>
+        </arguments>
+        <grabMultiple selector="{{productSelector}}" stepKey="grabProducts"/>
+    </actionGroup>
+
+    <actionGroup name="VerifyProductCount" extends="GetProductCount">
+        <arguments>
+            <argument name="count" type="string"/>
+        </arguments>
+        <assertCount stepKey="assertCount" after="grabProducts">
+            <expectedResult type="int">{{count}}</expectedResult>
+            <actualResult type="variable">grabProducts</actualResult>
+        </assertCount>
+    </actionGroup>
+</actionGroups>
+```
+
+The new action groups correspond to:
+
+```xml
+<actionGroups>
+    <actionGroup name="GetProductCount">
+        <arguments>
+            <argument name="productSelector" type="string"/>
+        </arguments>
+        <grabMultiple selector="{{productSelector}}" stepKey="grabProducts"/>
+    </actionGroup>
+
+    <actionGroup name="VerifyProductCount">
+        <arguments>
+            <argument name="count" type="string"/>
+            <argument name="productSelector" type="string"/>
+        </arguments>
+        <grabMultiple selector="{{productSelector}}" stepKey="grabProducts"/>
+        <assertCount stepKey="assertCount">
+            <expectedResult type="int">{{count}}</expectedResult>
+            <actualResult type="variable">grabProducts</actualResult>
+        </assertCount>
+    </actionGroup>
+</actionGroups>
+```
+
+## Extend Data
+
+### Update a data entry
```

Comment:
comment from @KevinBKozan:
The type attribute in data entities is pretty important, we should include its use in the xml examples according to how we allow type to be used or not used in conjunction with extends.